### ISSUE TITLE
Sprite `animation_done` different behavior on different platforms

### ISSUE THREAD

**Issue by Dragosha** [defold/defold#7263]
**Describe the bug (REQUIRED)**

There is a sprite in the atlas without any animation. When we call sprite.play_flipbook we have catch an `animation_done` message on Windows platform, but no any message on HTML5 platform.

```
function init(self)
	print("1. play_flipbook")
	sprite.play_flipbook("#sprite", "logo_blue_256")
end

function on_message(self, message_id, message, sender)
	if message_id == hash("animation_done") then
		print("2. animation_done", sender)
	end
end
```

Windows debug:
```
DEBUG:SCRIPT: 1. play_flipbook
DEBUG:SCRIPT: 2. animation_done	url: [main:/go#sprite]
```

HTML5:
`DEBUG:SCRIPT: 1. play_flipbook
`

**Expected behavior (REQUIRED)**
The same behavior on different platforms. And no `animation_done` is more clear in case when sprite doesn't have animation sequence.

In my case, I encountered this difference in behavior when I tried to set images to factory spawned objects when the main object was destroyed.  In this case it can't accept the message from engine or the callback from sprite.play_flipbook method with a corresponding message in the console.
Code example: 
```
...
for i = 1, math.random(4, 6) do
    local id = factory.create("#part_factory", position)
    local s = msg.url(nil, id, "sprite")
    sprite.play_flipbook(s, "part_"..math.random(4))
...
end
go.delete()
...

```

**Defold version (REQUIRED):**
  - Version [e.g. 1.4.1]

**Minimal repro case project (OPTIONAL):**
[animation_done.zip](https://github.com/defold/defold/files/10379676/animation_done.zip)

**Comment by britzl** [comment]
Furthermore this does only happen if there is no animation explicitly added to the atlas. If I add a single image "once forward" animation and use that animation when calling sprite.play_flipbook() I get the same result in bundles and builds, with both 1 and 2 showing.

**Comment by britzl** [comment]
With a bundle using bob the playback for an implicit animation (ie single image) will be set to NONE, which results in the animation not playing, and thus no `animation_done` message.

**Comment by britzl** [comment]
Fixed by #7542
